core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -787,9 +787,10 @@ class DAGScheduler(
       }
     }
 
+    // Create internal accumulators if the stage has no accumulators initialized.
     // Reset internal accumulators only if this stage is not partially submitted
     // Otherwise, we may override existing accumulator values from some tasks
-    if (allPartitions == partitionsToCompute) {
+    if (stage.internalAccumulators.isEmpty || allPartitions == partitionsToCompute) {
```

Comment:
@andrewor14 we don't actually care about clearing the values, right?  Can we simplify this to just `if (stage.internalAccumulators.isEmpty)`? Or just initialize them when a `Stage` is constructed?